runtime-core/core-types/decl/optional.h

```diff
@@ -203,3 +203,17 @@ using enable_if_t_is_optional_t2 = std::enable_if_t<std::is_same<T, Optional<T2>
 
 template<class T>
 using enable_if_t_is_optional_string = enable_if_t_is_optional_t2<T, string>;
+
+template<typename T>
+struct InternalOptionalType {
+  using type = T;
+};
+
+template<typename T>
+struct InternalOptionalType<Optional<T>> {
+  using type = T;
+};
+
+template<typename T>
+using internal_optional_type = typename InternalOptionalType<T>::type;
```

Comment:
Let's name it `internal_optional_type_t` in a manner that standard lib names such traits